profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfilePrivacyLogicImpl.java

```diff
@@ -34,76 +32,40 @@
  * @author Steve Swinsburg (steve.swinsburg@gmail.com)
  *
  */
+@Setter
 @Slf4j
 public class ProfilePrivacyLogicImpl implements ProfilePrivacyLogic {
 
-	private Cache cache;
-	private final String CACHE_NAME = "org.sakaiproject.profile2.cache.privacy";
+	private ProfileConnectionsLogic connectionsLogic;
+	private ProfileDao dao;
+	private SakaiProxy sakaiProxy;
 
-	/**
- 	 * {@inheritDoc}
- 	 */
 	@Override
 	public ProfilePrivacy getPrivacyRecordForUser(final String userId) {
-		return getPrivacyRecordForUser(userId, true);
-	}
-	
-	/**
- 	 * {@inheritDoc}
- 	 */
-	@Override
-	public ProfilePrivacy getPrivacyRecordForUser(final String userId, final boolean useCache) {
-		
-		if(userId == null){
-			throw new IllegalArgumentException("Null argument in ProfileLogic.getPrivacyRecordForUser"); 
-		}
 		
-		//will stay null if we can't get or create one
-		ProfilePrivacy privacy = null;
-		
-		//check cache
-		if(useCache) {
-			if(cache.containsKey(userId)){
-				log.debug("Fetching privacy record from cache for: " + userId);
-				privacy = (ProfilePrivacy)cache.get(userId);
-				if(privacy != null) {
-				  return(privacy);
-				}
-				// This means that the cache has expired. evict the key from the cache
-				log.debug("Privacy cache appears to have expired for " + userId);
-				this.cacheManager.evictFromCache(this.cache, userId);
-			}
+		if (StringUtils.isBlank(userId)) {
+			throw new IllegalArgumentException("Invalid userid " + userId);
 		}
 		
-		privacy = dao.getPrivacyRecord(userId);
-		log.debug("Fetching privacy record from dao for: " + userId);
+		// will stay null if we can't get or create one
+		ProfilePrivacy privacy = dao.getPrivacyRecord(userId);
+		log.debug("Fetching privacy record from dao for [{}]", userId);
 		
-		//if none, create and persist a default
-		if(privacy == null) {
+		// if none, create and persist a default
+		if (privacy == null) {
 			privacy = dao.addNewPrivacyRecord(getDefaultPrivacyRecord(userId));
-			if(privacy != null) {
-				sakaiProxy.postEvent(ProfileConstants.EVENT_PRIVACY_NEW, "/profile/"+userId, true);
-				log.info("Created default privacy record for user: " + userId); 
-			}
-		}
-		
-		//add to cache
-		if(privacy != null) {
-			log.debug("Adding privacy record to cache for: " + userId);
-			cache.put(userId, privacy);
 		}
-		
-		//if still null, we can't do much except log an error and wait for an NPE.
-		if(privacy == null) {
-			log.error("Couldn't retrieve or create a privacy record for user: " + userId + " This is an error and you need to fix your installation.");
+
+		if (privacy == null) {
```

Comment:
This dao.addNewPrivacyRecord(getDefaultPrivacyRecord(userId)); could return null?

Reply:
correct that is how it was previously